File: schemaorg/cancelaction.go
package schemaorg

import "encoding/json"

// CancelAction see : https://schema.org/CancelAction
type CancelAction struct {

typeContext

PlanAction

// ScheduledTime see : /scheduledTime
// The time the object is scheduled to.
ScheduledTime interface{} `json:"scheduledTime"`

}

func (v *CancelAction) MarshalJSON() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "CancelAction"

	return json.Marshal(v)
}

File: schemaorgphp/classes/cancel_action.php
<?php

class CancelAction extends PlanAction implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'CancelAction';
	
	/**
	 * The time the object is scheduled to.
	 * see : https://schema.org/scheduledTime
	 * @var string|string[]
	 */
	public var $scheduled_time;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'CancelAction'
		);
		
		$serialized = so_json_serialize( $this->scheduled_time );
		if ( ! empty( $serialized ) ) {
			$out['scheduledTime'] = $serialized;
		}
		
		return $out;
	}
}
